packages/react-reconciler/src/ReactFiber.old.js

```diff
@@ -413,6 +415,8 @@ export function resetWorkInProgress(
     workInProgress.memoizedProps = current.memoizedProps;
     workInProgress.memoizedState = current.memoizedState;
     workInProgress.updateQueue = current.updateQueue;
+    // Needed because Blocks store data on type.
+    workInProgress.type = current.type;
```

Comment:
I don't have coverage for the SuspenseList codepath but I figured it would need to be updated too.